src/tsconfig-loader.ts

```diff
@@ -90,25 +90,26 @@ function resolveConfigPath(cwd: string, filename?: string): string | undefined {
   const configAbsolutePath = walkForTsConfig(cwd);
   return configAbsolutePath ? path.resolve(configAbsolutePath) : undefined;
 }
-
 export function walkForTsConfig(
   directory: string,
-  // eslint-disable-next-line no-shadow
-  existsSync: (path: string) => boolean = fs.existsSync
+  readdirSync: (path: string) => string[] = fs.readdirSync
 ): string | undefined {
-  const configPath = path.join(directory, "./tsconfig.json");
-  if (existsSync(configPath)) {
-    return configPath;
+  const files = readdirSync(directory);
+  const filesToCheck = ["tsconfig.json", "jsconfig.json"];
+  for (const fileToCheck of filesToCheck) {
+    if (files.indexOf(fileToCheck) !== -1) {
+      return path.join(directory, fileToCheck);
+    }
   }
 
-  const parentDirectory = path.join(directory, "../");
```

Comment:
@jonaskello Was there a good reason for using `path.join` instead of `path.dirname` here?

Reply:
* `path.join('/root/dir', '../')` return `/root/`
* `path.dirname('/root/dir')` return `/root`(without the trailing directory separators)

the latter is normalized, easier to mock when we mock the `fs.readdirSync`, so I adjust `path.join` to `path.dirname`; though `path.join(directory, '..')` is fine either.